### PR TITLE
MAGN-5849: Custom node created from selection does not get chance to be saved when exiting Dynamo

### PR DESCRIPTION
**Pull request by pboyer**
### Issue 

A custom node created from selection is never added to the workspace collection.  As a result, the user never gets the chance to save it.  

### Fix

Add it to the visible workspace collection!  I also made this collection read only by changing it to an IEnumerable property (as it should be).  

### Tests

I added a test to reproduce. All of the other custom node tests continue to work.  

### Reviewer

@ikeough

### CODE REVIEW

File: src/DynamoCore/Utilities/Extensions.cs [hunk 1]
```diff
@@ -29,6 +29,25 @@ public interface ITree<out TNodeTag, out TLeaf>
     public static class ExtensionMethods
     {
         /// <summary>
+        /// Get the index of an element in an IEnumerable
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="source">The IEnumerable</param>
+        /// <param name="value">The value for which the index is sought</param>
+        /// <returns>Zero or greater if in the IEnumerable, otherwise -1</returns>
+        public static int IndexOf<T>(this IEnumerable<T> source, T value)
```
Comment (ikeough) [inline src/DynamoCore/Utilities/Extensions.cs]: Where is this new extension method used? Did you make this then realize IEnumerable has ElementAt?

File: src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs [hunk 2]
```diff
@@ -966,8 +966,9 @@ internal void CollapseNodes(IEnumerable<NodeModel> selectedNodes)
             if (!args.Success)
                 return;
 
-            DynamoViewModel.Model.CustomNodeManager.Collapse(
-                selectedNodes, Model, DynamoModel.IsTestMode, args);
+            DynamoViewModel.Model.AddCustomNodeWorkspace(
```
Comment (ikeough) [inline src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs]: What happens when Collapse fails? Is it safe to make this call here?

**Comment by ikeough** [comment]
A couple of small comments, then LGTM.

File: src/DynamoCore/Utilities/Extensions.cs (hunk 1 above)
Comment (pboyer) [inline src/DynamoCore/Utilities/Extensions.cs]: ElementAt is for indexing.  It's different from this.

Comment (pboyer) [inline src/DynamoCore/Utilities/Extensions.cs]: `List<T>` has this method.  There's a method on the workspace that allows you to get the index from a Workspace argument.  This preserves that method which is out of scope.

**Comment by pboyer** [comment]
Thanks @ikeough.  I made some small changes for your comments.
